airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/JobNotifier.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.scheduler.persistence;
+
+import io.airbyte.config.Notification;
+import io.airbyte.config.Notification.NotificationType;
+import io.airbyte.config.StandardDestinationDefinition;
+import io.airbyte.config.StandardSourceDefinition;
+import io.airbyte.config.StandardWorkspace;
+import io.airbyte.config.persistence.ConfigNotFoundException;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.config.persistence.PersistenceConstants;
+import io.airbyte.notification.NotificationClient;
+import io.airbyte.notification.SlackNotificationClient;
+import io.airbyte.scheduler.models.Job;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JobNotifier {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JobNotifier.class);
+
+  public static final String NOTIFICATION_TEST_MESSAGE = "Hello World! This is a test trying to send a message from Airbyte...";
+
+  private final String connectionPageUrl;
+  private final ConfigRepository configRepository;
+
+  public JobNotifier(String webappUrl, ConfigRepository configRepository) {
+    if (webappUrl.endsWith("/")) {
+      this.connectionPageUrl = String.format("%ssource/connection/", webappUrl);
+    } else {
+      this.connectionPageUrl = String.format("%s/source/connection/", webappUrl);
+    }
+    this.configRepository = configRepository;
+  }
+
+  public boolean sendTestNotifications(final StandardWorkspace workspace) {
```

Comment:
I'm adding this method to be called by the API as a test (while setting up the user's preferences for example) for this notification...

Reply:
This shouldn't exist there. For the test you don't want to test the code logic. just the delivery so you should only use a NotificationClient